### ISSUE TITLE
OpenSSL v1.1.0 build issues

### ISSUE THREAD

**Issue by zwoop** [apache/trafficserver#1767]
Getting two errors (well, one error and one warning):

```
  CXX      experimental/sslheaders/experimental_sslheaders_libsslhdr_la-expand.lo
experimental/sslheaders/expand.cc: In function 'void x509_expand_signature(X509*, BIO*)':
experimental/sslheaders/expand.cc:73:30: error: invalid use of incomplete type 'X509 {aka struct x509_st}'
   ASN1_BIT_STRING *sig = x509->signature;
                              ^~
In file included from experimental/sslheaders/expand.cc:19:0:
experimental/sslheaders/sslheaders.h:26:16: note: forward declaration of 'X509 {aka struct x509_st}'
 typedef struct x509_st X509;
                ^~~~~~~
Makefile:4625: recipe for target 'experimental/sslheaders/experimental_sslheaders_libsslhdr_la-expand.lo' failed
gmake[2]: [experimental/sslheaders/experimental_sslheaders_libsslhdr_la-expand.lo] Error 1 (ignored)
```

```
experimental/ssl_cert_loader/ssl-cert-loader.cc: In function 'SSL_CTX* {anonymous}::Load_Certificate(const {anonymous}::SslEntry*, std::deque<std::basic_string<char> >&)':
experimental/ssl_cert_loader/ssl-cert-loader.cc:201:36: warning: 'unsigned char* ASN1_STRING_data(ASN1_STRING*)' is deprecated [-Wdeprecated-declarations]
           char *name_ptr = (char *)ASN1_STRING_data(alt_name->d.dNSName);
                                    ^~~~~~~~~~~~~~~~
In file included from /opt/openssl/include/openssl/bn.h:31:0,
                 from /opt/openssl/include/openssl/asn1.h:24,
                 from /opt/openssl/include/openssl/objects.h:916,
                 from /opt/openssl/include/openssl/evp.h:27,
                 from /opt/openssl/include/openssl/x509.h:23,
                 from /opt/openssl/include/openssl/ssl.h:50,
                 from experimental/ssl_cert_loader/ssl-cert-loader.cc:30:
/opt/openssl/include/openssl/asn1.h:553:35: note: declared here
 DEPRECATEDIN_1_1_0(unsigned char *ASN1_STRING_data(ASN1_STRING *x))
                                   ^
/opt/openssl/include/openssl/opensslconf.h:102:35: note: in definition of macro 'DECLARE_DEPRECATED'
 # define DECLARE_DEPRECATED(f)    f __attribute__ ((deprecated));
                                   ^
/opt/openssl/include/openssl/asn1.h:553:1: note: in expansion of macro 'DEPRECATEDIN_1_1_0'
 DEPRECATEDIN_1_1_0(unsigned char *ASN1_STRING_data(ASN1_STRING *x))
 ^~~~~~~~~~~~~~~~~~
experimental/ssl_cert_loader/ssl-cert-loader.cc:201:72: warning: 'unsigned char* ASN1_STRING_data(ASN1_STRING*)' is deprecated [-Wdeprecated-declarations]
           char *name_ptr = (char *)ASN1_STRING_data(alt_name->d.dNSName);
                                                                        ^
In file included from /opt/openssl/include/openssl/bn.h:31:0,
                 from /opt/openssl/include/openssl/asn1.h:24,
                 from /opt/openssl/include/openssl/objects.h:916,
                 from /opt/openssl/include/openssl/evp.h:27,
                 from /opt/openssl/include/openssl/x509.h:23,
                 from /opt/openssl/include/openssl/ssl.h:50,
                 from experimental/ssl_cert_loader/ssl-cert-loader.cc:30:
/opt/openssl/include/openssl/asn1.h:553:35: note: declared here
 DEPRECATEDIN_1_1_0(unsigned char *ASN1_STRING_data(ASN1_STRING *x))
                                   ^
/opt/openssl/include/openssl/opensslconf.h:102:35: note: in definition of macro 'DECLARE_DEPRECATED'
 # define DECLARE_DEPRECATED(f)    f __attribute__ ((deprecated));
                                   ^
/opt/openssl/include/openssl/asn1.h:553:1: note: in expansion of macro 'DEPRECATEDIN_1_1_0'
 DEPRECATEDIN_1_1_0(unsigned char *ASN1_STRING_data(ASN1_STRING *x))
 ^~~~~~~~~~~~~~~~~~
```

**Comment by zwoop** [comment]
Found this from the curl repo, which seems to address similar issues (around X509):

https://github.com/curl/curl/commit/7f683b0ea87ed158eb4bd22cdd1f26eb901a97d0

**Comment by zwoop** [comment]
Oh, I think this is fixed in 22994b0dc3ad5827a26a6fed2ee525cc16b476d8, maybe we need to cherry pick this.

**Comment by zwoop** [comment]
This is fixed.
